pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -119,23 +105,120 @@ func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRe
 // name.
 // Returns an error if any error is encountered while attempting to
 // read the previously persisted information from the API server.
-// If no previous information has been persisted, it returns an empty object.
-func (or *operationRequestStoreOnETCD) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
+func (or *operationRequestStore) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
 	log := logger.GetLogger(ctx)
-	// TODO: Implement GetRequestDetails for operationRequestStoreOnETCD
-	err := fmt.Errorf("GetRequestDetails not implemented for operationRequestStoreOnETCD")
-	log.Error(err)
-	return nil, err
+	instanceKey := client.ObjectKey{Name: name, Namespace: csiconfig.DefaultCSINamespace}
+	log.Infof("Getting CnsVolumeOperationRequest instance with name %s/%s", instanceKey.Namespace, instanceKey.Name)
+
+	instance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{}
+	err := or.k8sclient.Get(ctx, instanceKey, instance)
+	if err != nil {
+		log.Errorf("failed to get CnsVolumeOperationRequest instance %s/%s with error: %v", instanceKey.Namespace, instanceKey.Name, err)
+		return nil, err
+	}
+	log.Debugf("Found CnsVolumeOperationRequest instance %v", spew.Sdump(instance))
+
+	// Determine which operation details need to be returned. Callers only need to know about the
+	// last operation that was invoked on a volume.
+	var operationDetailsToReturn cnsvolumeoperationrequestv1alpha1.OperationDetails
+	if len(instance.Status.LatestOperationDetails) != 0 {
+		operationDetailsToReturn = instance.Status.LatestOperationDetails[len(instance.Status.LatestOperationDetails)-1]
+	}
```

Comment:
Ideally this situation shouldnt exist since the instances are created with atleast 1 item in `LatestOperationDetails` . By similar do you mean `NotFound` error?

Reply:
A general error is also fine as I dont know if the caller would care about the type of the error. How does the caller code looks like?

Reply:
if we didnt return an error here, the caller would assume that the operation has not been performed previously (because there's no task id returned) and attempt to perform that operation. 
but if we do return an error, then i suppose the callers are going to be stuck in a failure loop

Reply:
@SandeepPissay modified to return an error